biggraphite/drivers/elasticsearch.py

```diff
@@ -0,0 +1,391 @@
+
+    _UUID_NAMESPACE = uuid.UUID("{00000000-1111-2222-3333-444444444444}")
+
+    def __init__(
+        self,
+        hosts=DEFAULT_HOSTS,
+        port=DEFAULT_PORT,
+        index=DEFAULT_INDEX,
+        username=None,
+        password=None,
+        timeout=DEFAULT_TIMEOUT,
+    ):
+        """Create a new ElasticSearchAccessor."""
+        super(_ElasticSearchAccessor, self).__init__("ElasticSearch")
+        self._metric_to_points = collections.defaultdict(sortedcontainers.SortedDict)
+        self._name_to_metric = {}
+        self._directory_names = sortedcontainers.SortedSet()
+        self.__downsampler = _downsampling.Downsampler()
+        self.__delayed_writer = _delayed_writer.DelayedWriter(self)
+        self._hosts = list(hosts)
+        self._port = port
+        # TODO: Have indices with date suffixes.
+        self._index = index
+        self._username = username
+        self._password = password
+        self._timeout = timeout
+        self.client = None
+
+    @staticmethod
+    def _components_from_name(metric_name):
+        res = metric_name.split(".")
+        return list(filter(None, res))
+
+    @property
+    def _metric_names(self):
+        return self._name_to_metric.keys()
+
+    def connect(self, *args, **kwargs):
+        """See the real Accessor for a description."""
+        super(_ElasticSearchAccessor, self).connect(*args, **kwargs)
+        self._connect()
+        self.is_connected = True
+
+    def _connect(self):
+        """Connect to elasticsearch."""
+        if self.is_connected:
+            return
+
+        if self._username:
+            http_auth = (self._username, self._password or "")
+        else:
+            http_auth = None
+        es = elasticsearch.Elasticsearch(
+            self._hosts,
+            port=self._port,
+            http_auth=http_auth,
+            # FIXME: set to true
+            sniff_on_start=False,
+            sniff_on_connection_fail=False,
+            retry_on_timeout=True,
+            max_retries=3,
+            timeout=self._timeout,
+        )
+
+        log.info("Connected: %s" % es.info())
+        self.client = es
+
+    def shutdown(self, *args, **kwargs):
+        """See the real Accessor for a description."""
+        super(_ElasticSearchAccessor, self).shutdown(*args, **kwargs)
+        self._shutdown()
+        self.is_connected = False
+
+    def _shutdown(self):
+        """Shutdown Elasticsearch client."""
+        if self.client:
+            self.client.transport.close()
+            self.client = None
+
+    def background(self):
+        """Perform periodic background operations."""
+        if self.__downsampler:
+            self.__downsampler.purge()
+        if self.__delayed_writer:
+            self.__delayed_writer.write_some()
+
+    def flush(self):
+        """Flush any internal buffers."""
+        if self.__delayed_writer:
+            self.__delayed_writer.flush()
+        self.client.fush(
```

Comment:
/fush/flush/

Reply:
done

Reply:
afaik you need client.indices.flush() according to https://elasticsearch-py.readthedocs.io/en/master/api.html#elasticsearch
'Elasticsearch' object has no attribute 'flush'

Reply:
fixed